src/components/UncontrolledTable.js

```diff
@@ -129,6 +131,34 @@ export default class UncontrolledTable extends React.Component {
     return isEqual(keys, newKeys);
   }
 
+  getVisibleRows = () => {
+    const { page } = this.state;
+    const { ascending, column } = this.state.sort;
+    const { paginated, pageSize, rows } = this.props;
+
+    const start = page * pageSize;
+    const end = start + pageSize;
+    const sortedRows = this.sortedData(rows, column, ascending);
+    return paginated ? sortedRows.slice(start, end) : sortedRows;
+  }
+
+  visibleRowsChanged(prevProps, prevState) {
+    const { page } = this.state;
+    const { ascending, column } = this.state.sort;
+    const { paginated, pageSize, rows } = this.props;
+
+    const { page: prevPage } = prevState;
+    const { ascending: prevAscending, column: prevColumn } = prevState.sort;
+    const { paginated: prevPaginated, pageSize: prevPageSize, rows: prevRows } = prevProps;
+
+    return page !== prevPage ||
+      ascending !== prevAscending ||
+      column !== prevColumn ||
+      paginated !== prevPaginated ||
+      pageSize !== prevPageSize ||
```

Comment:
This is a good example of where converting to a functional component could make this a bit easier, which you don't need to do now!

But for an example, this would be the way with hooks:
```
useEffect(() => {
  onVisibleRowsChange(getVisibleRows());
}, [page, ascending, column, paginated, pageSize, rows])
```
Or something similar to the above.

Reply:
I 100% agree. Converting to a functional component was my first impulse (followed closely by converting to TS)!